File: analytics_assistant/src/api/models/session.py
# -*- coding: utf-8 -*-
"""
会话相关 Pydantic 模型

定义会话 CRUD 的请求和响应数据结构。
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from .chat import Message

class CreateSessionRequest(BaseModel):
    """创建会话请求。"""

    title: Optional[str] = Field(None, description="会话标题")

class CreateSessionResponse(BaseModel):
    """创建会话响应。"""

    session_id: str = Field(..., description="会话 ID")
    created_at: datetime = Field(..., description="创建时间")

class SessionResponse(BaseModel):
    """会话详情模型。"""

    id: str
    tableau_username: str
    title: str
    messages: list[Message]
    created_at: datetime
    updated_at: datetime

class GetSessionsResponse(BaseModel):
    """获取会话列表响应。"""

    sessions: list[SessionResponse]
    total: int

class UpdateSessionRequest(BaseModel):
    """更新会话请求。"""

    title: Optional[str] = None
    messages: Optional[list[Message]] = None

File: analytics_assistant/frontend/src/api/session.ts
import apiClient from './client'

export interface SessionMessage {
  role: 'user' | 'assistant' | 'system'
  content: string
  created_at?: string
}

export interface ApiSession {
  id: string
  tableau_username: string
  title: string
  messages: SessionMessage[]
  created_at: string
  updated_at: string
}

export interface SessionListResponse {
  sessions: ApiSession[]
  total: number
}

export interface CreateSessionRequest {
  title?: string
}

export interface UpdateSessionRequest {
  title?: string
  messages?: SessionMessage[]
}

export const sessionApi = {
  async getSessions(params?: { offset?: number; limit?: number }): Promise<SessionListResponse> {
    return apiClient.get<SessionListResponse>('/api/sessions', { params })
  },

  async getSession(sessionId: string): Promise<ApiSession> {
    return apiClient.get<ApiSession>(`/api/sessions/${sessionId}`)
  },

  async createSession(data: CreateSessionRequest): Promise<{ session_id: string; created_at: string }> {
    return apiClient.post<{ session_id: string; created_at: string }>('/api/sessions', data)
  },

  async updateSession(sessionId: string, data: UpdateSessionRequest): Promise<ApiSession> {
    return apiClient.put<ApiSession>(`/api/sessions/${sessionId}`, data)
  },

  async deleteSession(sessionId: string): Promise<void> {
    await apiClient.delete(`/api/sessions/${sessionId}`)
  },
}
